crates/factor-outbound-mqtt/src/host.rs

```diff
@@ -0,0 +1,131 @@
+use std::{sync::Arc, time::Duration};
+
+use anyhow::Result;
+use spin_core::{async_trait, wasmtime::component::Resource};
+use spin_factor_outbound_networking::OutboundAllowedHosts;
+use spin_world::v2::mqtt::{self as v2, Connection, Error, Qos};
+use tracing::{instrument, Level};
+
+pub type CreateClient = Arc<
```

Comment:
Looking at the other factors, we've normally been handling this pattern with custom trait trait objects rather than closure trait objects. So instead of this type alias we would do something like:


```rust
trait ClientCreator {
  fn create(
    address: String,
    username: String,
    password: String,
    keep_alive_interval: Duration,
  )
}
// ...

impl InstanceState {
    pub fn new(allowed_hosts: OutboundAllowedHosts, create_client: Box<dyn ClientCreator>) { 
// ...
```